packcheck.sh

```diff
@@ -1013,47 +1013,97 @@ ensure_cabal() {
       die "Cabal version should at least be $MIN_CABALVER"
 }
 
+# Find a file in the parent/ancestor directories
+find_in_parents () {
+  local curdir="$1" # must be absolute path without ..
+  local file="$2"
+
+  if test -e "$curdir/$file"
+  then echo "$curdir/$file"
+  elif test "$curdir" != "/"
+  then find_in_parents "$(dirname "$curdir")" "$file"
+  fi
+}
+
 ensure_cabal_project() {
   CABALCMD=cabal
   SDIST_CABALCMD=$CABALCMD
 
   if test -n "$CABAL_PROJECT"
   then
-    require_file $CABAL_PROJECT
+    require_file "$CABAL_PROJECT"
     SDIST_CABALCMD="$CABALCMD --project-file=$CABAL_PROJECT"
     if test -n "$DISABLE_SDIST_BUILD"
-    then
+    then # regular build
       CABALCMD="$SDIST_CABALCMD"
-    else
+    else # sdist build
       # We run the command from .packcheck/<package> dir after unpacking
       # sdist
       CABALCMD="$CABALCMD --project-file=../../$CABAL_PROJECT"
     fi
   fi
-  echo "Using cabal command [$CABALCMD]"
-}
 
-ensure_stack_yaml() {
-  if test -n "$STACK_YAML"
-  then
-    require_file $STACK_YAML
+  if test -n "$CABAL_PROJECT"
```

Comment:
I wonder if you can merge this with the upper conditional.

Reply:
Probably yes. Earlier I was adding more code here, that's why it was separate. And it was easier to think by keeping the old code intact.

Reply:
Merged.